Partially prepare grp/classic.gi for MatrixObj

fingolfin opened the pull request:
- use mat[i,j] instead of mat[i][j]
- avoid using Length(mat)
- consistently write `Sp(n,q)` instead of `SP(n,q)`
- use NullMat / IdentityMat instead of tricks using `List` and `ShallowCopy`


Ideally should use ZeroMatrix / IdentityMatrix, but there are some caveats with those right now, so I am doing this intermediate step now, while working on another matobj related PR.

ThomasBreuer reviewed (approved):
This looks good.
There is still one instance of `List( ..., ShallowCopy )` in `grp/classic.gi`.

fingolfin commented:
Indeed; I removed that last use of `ShallowCopy`, and then noticed two uses of `StructuralCopy` in `WreathProductOfMatrixGroup`, and *then* I noticed that there now were a bunch of variables named `id` that got defined but never used, so I removed those as well.
